### PR TITLE
Bump k/release to v0.6.0

### PR DESCRIPTION
**Pull request by saschagrunert**
Signed-off-by: Sascha Grunert <sgrunert@suse.com>

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -375,6 +375,7 @@ jobs:
             - bd:a6:4b:50:1b:2a:a4:1d:79:67:64:05:66:c4:3a:10
       - run:
           name: Generate Release Notes
+          no_output_timeout: 30m
```
Comment (kolyshkin) [inline .circleci/config.yml:378]: Maybe move to a separate commit (so there's a chance to have a commit message explaining why this change is done)?

**Comment by kolyshkin** [comment]
Did you do it by
```
go get k8s.io/release@v0.6.0
make vendor
# fix scripts/* to work with updated k/release
```

or was it anything else?

**Comment by kolyshkin** [comment]
Needs a rebase (probably because #4361 has been merged recently)

File: .circleci/config.yml (hunk 1 above)
Comment (saschagrunert) [reply .circleci/config.yml:378]: Yep, adjusted as suggested.

**Comment by saschagrunert** [comment]
> Did you do it by
> 
> ```
> go get k8s.io/release@v0.6.0
> make vendor
> # fix scripts/* to work with updated k/release
> ```
> 
> or was it anything else?

What I usually do is:

1. Modify go.mod by only bumping k/release to v0.6.0
2. Run `make vendor GOSUMDB=`

This also bumps other required dependencies, too.

**Comment by saschagrunert** [comment]
Had to rebase, PTAL @haircommander
